FeedStoreChallenge/Helpers/CoreDataHelpers.swift

```diff
@@ -26,3 +26,42 @@ extension NSManagedObjectModel {
 		self.init(contentsOf: momd)
 	}
 }
+
+extension ManagedFeedImage {
+	static func create(with feedImage: LocalFeedImage, in context: NSManagedObjectContext) -> ManagedFeedImage {
+		let image = ManagedFeedImage(context: context)
+		image.id = feedImage.id
+		image.location = feedImage.location
+		image.url = feedImage.url
+		image.desc = feedImage.description
+		return image
+	}
+
+	func toLocalFeedImage() -> LocalFeedImage {
+		return LocalFeedImage(id: id, description: desc, location: location, url: url)
+	}
+}
+
+extension ManagedCache {
+	@discardableResult
+	static func create(with feed: [LocalFeedImage], timestamp: Date, in context: NSManagedObjectContext) -> ManagedCache {
+		let cache = ManagedCache(context: context)
+		let images = feed.map { ManagedFeedImage.create(with: $0, in: context) }
+		cache.images = NSOrderedSet(array: images)
+		cache.timestamp = timestamp
+		return cache
+	}
+
+	static func find(in context: NSManagedObjectContext) throws -> [ManagedCache]? {
+		let cacheFetch: NSFetchRequest = ManagedCache.fetchRequest()
+		return try context.fetch(cacheFetch)
+	}
```

Comment:
Optional return type can be confusing here. If there is a cache then CoreData would return an array. If there is no cache then CoreData would return an empty array. If there is an error then the method would throw. There is no case where CoreData could return `nil`. `context.fetch` returns a non-optional array.

Please make the return array non-optional

```suggestion
	static func find(in context: NSManagedObjectContext) throws -> [ManagedCache] {
		let cacheFetch: NSFetchRequest = ManagedCache.fetchRequest()
		return try context.fetch(cacheFetch)
	}
```